src/router/mod.rs

```diff
@@ -120,18 +120,8 @@ impl<Route: 'static + Debug + PartialEq + ParsePath + Default + Clone + Navigati
         self.update_data(|data| data.push_to_history(route));
     }
 
-    /// Go back in history and navigate back to the previous route.
-    ///  # Note for now it does not add to history since we navigate inside
-    pub fn back(&self) -> bool {
-        self.can_back_with_route().map_or(false, |next_route| {
-            self.set_current_route(&next_route);
-            self.update_data(|data| data.current_history_index -= 1);
-            true
-        })
-    }
-
-    /// Check if you can go back in history and give you the previous route.
-    pub fn can_back_with_route(&self) -> Option<Route> {
+    /// If a previous `Route` in history exists, return it. Otherwise return `None`
```

Comment:
I've moved the order in the file that these functions are in to be as follows

1. `peek_back`
2. `peek_forward`
3. `back`
4. `forward`

the peek functions just give info. the navigation functions "do the giving of info, plus that other thing as well". A user can read the documentation, learn what the peek does, then how the navigation thing is an extension of the peek.